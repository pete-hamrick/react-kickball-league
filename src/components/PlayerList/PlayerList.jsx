import React from 'react'

export default function PlayerList() {
  return (
    <section>
      <h1>Player List View</h1>
    </section>
  )
}
