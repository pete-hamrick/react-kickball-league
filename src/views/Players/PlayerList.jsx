import React from 'react'
import { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { getPlayers } from '../../services/players'

export default function PlayerList() {
  const [players, setPlayers] = useState([])
  const [loading, setLoading] = useState([])

  useEffect(() => {
    async function getAllPlayers() {
      const data = await getPlayers()
      setPlayers(data)
      setLoading(false)
    }

    getAllPlayers()
  }, [])

  //TODO write handleDelete
  const handleDelete = async () => {}

  if (loading) return <h1>Loading Players...</h1>

  return (
    <section>
      <p>
        <Link to="/">Back to Homepage</Link>
        <Link to="/players/new">Add New Player</Link>
      </p>
      <ul className="players-list" aria-label="Player List">
        {players.map((player) => (
          <li key={player.id}>
            {player.name}
            <button>
              <Link to={`/players/${player.id}`}>View</Link>
            </button>
            <button>
              <Link to={`/players/update/${player.id}`}>Update</Link>
            </button>
            <button onClick={() => handleDelete(player)}>Delete</button>
          </li>
        ))}
      </ul>
    </section>
  )
}
