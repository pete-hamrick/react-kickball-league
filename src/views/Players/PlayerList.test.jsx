import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom'
import PlayerList from './PlayerList'

it('should render a list of the players', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/players']}>
      <Route path="/players">
        <PlayerList />
      </Route>
    </MemoryRouter>
  )

  screen.getByText('Loading Players...')

  const playerName = await screen.findByText('Johnny Baseball', { exact: false })

  expect(container).toMatchSnapshot()
  expect(playerName).toBeInTheDocument()
})
