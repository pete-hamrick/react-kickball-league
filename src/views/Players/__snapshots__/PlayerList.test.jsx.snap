// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of the players 1`] = `
<div>
  <section>
    <p>
      <a
        href="/"
      >
        Back to Homepage
      </a>
    </p>
    <ul
      aria-label="Player List"
      class="players-list"
    >
      <li>
        <a
          href="/players/1"
        >
          Johnny Baseball
        </a>
      </li>
    </ul>
  </section>
</div>
`;
